build/build-allocation-images/go/Dockerfile

```diff
@@ -0,0 +1,38 @@
+# Copyright 2019 Google LLC All Rights Reserved.
```

Comment:
This file looks identical to https://github.com/googleforgames/agones/blob/master/build/build-sdk-images/go/Dockerfile. Is there a reason we are creating a new docker image with exactly the same tooling installed instead of reusing the other one?

Reply:
You are so right. We need to do a refactoring on the SDK to move common files around and remove duplication. I didn't want to include refactoring and code generation in a single commit. There is another file build-sdk-images/tool/base that needs to be moved out of the sdk folder. I'll get to the refactoring in a future PR.

Reply:
Can you file a tracking issue for it? If you mark it as `help-wanted` then it's also possible that someone else will pick it up too. :)

Reply:
done https://github.com/googleforgames/agones/issues/1115